common/check.h

```diff
@@ -14,30 +14,35 @@ namespace Carbon {
 // a bug in the application.
 //
 // For example:
-//   CARBON_CHECK(is_valid) << "Data is not valid!";
-#define CARBON_CHECK(...)                                                   \
-  (__VA_ARGS__) ? (void)0                                                   \
-                : CARBON_CHECK_INTERNAL_STREAM()                            \
-                      << "CHECK failure at " << __FILE__ << ":" << __LINE__ \
-                      << ": " #__VA_ARGS__                                  \
-                      << Carbon::Internal::ExitingStream::AddSeparator()
+//   CARBON_CHECK(is_valid, "Data is not valid!");
+#define CARBON_CHECK(condition, ...) \
+  (condition) ? (void)0              \
+              : CARBON_INTERNAL_CHECK(condition __VA_OPT__(, ) __VA_ARGS__)
```

Comment:
This no longer supports commas in `condition`, which might be worth mentioning in the comment here. It looks (from a brief non-exhaustive skip) like this wasn't actually a problem for any of our existing checks, though?
```suggestion
//   CARBON_CHECK(is_valid, "Data is not valid!");
//
// The condition must be parenthesized if it contains top-level commas, for
// example in a template argument list:
//   CARBON_CHECK((inst.IsOneOf<Call, TupleLiteral>()),
//                "Unexpected inst {0}", inst);
#define CARBON_CHECK(condition, ...) \
  (condition) ? (void)0              \
              : CARBON_INTERNAL_CHECK(condition __VA_OPT__(, ) __VA_ARGS__)
```

Reply:
There were a few in `//explorer` that I had to add parentheses for, but generally was very rare. Added comment.